Test/Test/ConsoleReader.h

```diff
@@ -1,30 +1,13 @@
 #pragma once
-#include <iostream>
-#include <conio.h>
+
+const char esc = 27;
+const char enter = 13;
+const char backspace = 8;
 
 class ConsoleReader {
-	public:
+	private:
 		std::string buffer = "";
-		int readConsole() {
-			char esc = 27;
-			char enter = 13;
-			char backspace = 8;
-			char c;
-			do {
-				c = _getch();
-				if (c != enter && c != esc) {
-					if (c != backspace)
-						buffer += c;
-					else
-						buffer.pop_back();
-					_putch(c);
-				}
-				// can getchar even get esc?
-				else
-					break;
-			} while (true);
-			if (c == esc)
-				buffer = "";
-			return buffer.length();
-		};
+	public:
+		std::string getData();
+		int readConsole(std::string const &message);
```

Comment:
Is this prompt message? The name should reflect the usage. Just message is not clear how it will be used. 
BTW, look for Rober C Martin "Clean code". It covers this approach :)

Reply:
Is "prompt" good? 

> Rober C Martin "Clean code"

I will look into that later, thanks